ionoscloud/resource_dbaas_pgsql_database.go

```diff
@@ -0,0 +1,104 @@
+package ionoscloud
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	pgsql "github.com/ionos-cloud/sdk-go-dbaas-postgres"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/services/dbaas"
+)
+
+func resourceDbaasPgSqlDatabase() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceDbaasPgSqlDatabaseCreate,
+		UpdateContext: resourceDbaasPgSqlDatabaseUpdate,
+		ReadContext:   resourceDbaasPgSqlDatabaseRead,
+		DeleteContext: resourceDbaasPgSqlDatabaseDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceDbaasPgSqlDatabaseImporter,
+		},
+		Schema: map[string]*schema.Schema{
+			"cluster_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				// TODO -- not sure if ForceNew makes sense here since there is no way to modify
+				// the database, so the update method will be empty
+				ForceNew:         true,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.IsUUID),
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Description: "The databasename of a given database.",
+				Required:    true,
+			},
+			"owner": {
+				Type:        schema.TypeString,
+				Description: "The name of the role owning a given database.",
+				Required:    true,
+			},
+		},
+		Timeouts: &resourceDefaultTimeouts,
+	}
+}
+
+func resourceDbaasPgSqlDatabaseCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(SdkBundle).PsqlClient
+	clusterId := d.Get("cluster_id").(string)
+	name := d.Get("name").(string)
+	owner := d.Get("owner").(string)
+	request := pgsql.Database{
+		Properties: &pgsql.DatabaseProperties{},
+	}
+	request.Properties.Name = &name
+	request.Properties.Owner = &owner
+
+	database, _, err := client.CreateDatabase(ctx, clusterId, request)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("an error occured while creating the PgSql database named: %s inside the cluster with ID: %s, error: %w", name, clusterId, err))
+	}
+	d.SetId(*database.Id)
+	return resourceDbaasPgSqlDatabaseRead(ctx, d, meta)
```

Comment:
Just call SetDatabasePgSqlData to not have any potential read/write inconsistencies

Reply:
@cristiGuranIonos done in cc3d6928b915b5af4bcaf43fbd60d2847e407fa5